isis/src/base/apps/spiceinit/spiceinit.cpp

```diff
@@ -695,5 +700,9 @@ namespace Isis {
 
     delete sunPosTable;
     sunPosTable = NULL;
+
+    // Remove csminit data from the cube after successful run
+    icube->deleteGroup("CsmInfo");
+    icube->deleteBlob("String","CSMState");
```

Comment:
I had to put this in two places because the on failure revert logic is located in two places, one for local SPICE and one for remove SPICE.